### ISSUE TITLE
Update Nokogiri version from 1.5.11 to 1.6

### ISSUE THREAD

**Issue by tnimmo** [fog/fog#3099]
Are there any plans to upgrade Nokogiri to version 1.6?

I have forked fog and upgraded nokogiri to 1.6 and all seems to be working. Should I put this in as PR or are there other considerations I haven't thought about?

From the Nokogiri change logs (1.6.0.rc1):

"Ruby 1.9.2 and higher now required"
"Support for Ruby 1.8.7 and prior has been dropped"

Is this a problem?

**Comment by isabanin** [comment]
With fog being dependent on Nokogiri 1.5 it's currently not possible to use it in a Rails 4.2 application, which requires Nokogiri 1.6.

**Comment by geemus** [comment]
@isabanin thanks for the note. Maybe our hand is going to be forced here. I need to get on top of 2.0 plans anyway...

**Comment by tokengeek** [comment]
I think this ticket belongs in https://github.com/fog/fog-xml

I actually thought this was no longer the case and that 1.8.7 users were specifying it in their application `Gemfile`s as they have to for a number of other dependencies.

`fog` will need an update to the 1.8.7 Gemfile to avoid breaking on travis however.

**Comment by geemus** [comment]
Ah, that sounds a good deal more promising.

**Comment by vlasar** [comment]
Any update on being able to use ```fog``` with Rails 4.2?

Do we need to wait for 2.0 version or did I misunderstood?

Thank for all your work.

**Comment by plribeiro3000** [comment]
@viasar Since `ruby 1.8.7` will be dropped on `fog 2.0.0` we can't bump the `nokogiri` dependency until it happens.

@geemus Perhaps we should hurry a bit on `fog 2.0.0` to add compatibility with `rails 4.2.0`?

**Comment by vlasar** [comment]
@plribeiro3000 Ah, makes sense. Thank you for the clarification.

**Comment by tokengeek** [comment]
I don't think there's much holding us up (except for available time).

We can bump the version numbers and merge the PR. It's just making sure
all the critical things (if any) are done.

**Comment by agis-** [comment]
Any news on this one? It's really blocking us from going to Rails 4.2 too.

**Comment by plribeiro3000** [comment]
I think it past time to move on with this. Let me sync to see what we need to accomplish to be sure not to break anything else.

I will let you know once we do it.

Thanks for your support!

**Comment by agis-** [comment]
@plribeiro3000 I'd be happy to help with this in order to speed things up. Let me know :)

**Comment by tokengeek** [comment]
@plribeiro3000 - I think so too.

I think the best plan is for @geemus to release tomorrow. Then immediately we update the gemspec and documentations dropping 1.8.7

**Comment by plribeiro3000** [comment]
@tokengeek Shall we merge the open PR's then?

**Comment by tokengeek** [comment]
@plribeiro3000 - after another release I think, just so there is a clear end version to the 1.x series.

Not sure how up to date they are in case they need rebasing.

**Comment by plribeiro3000** [comment]
alright. One last release for the 1.x series then. Later today i will take a look and see if there is something to be updated but last time i checked we are good to go.

**Comment by tokengeek** [comment]
Removed the constraint, Travis passes and so I've merged.

Hopefully that removes this as a blocker for folks. :smile:

**Comment by agis-** [comment]
@tokengeek Great to hear! Will you also cut 2.0 soon or?

Thanks.

**Comment by tokengeek** [comment]
At the moment @geemus is the only one of us that release a new version.

I'm sure when he's awake in his time zone he can look at doing so. @plribeiro3000 might have some changes to merge in first based on https://github.com/fog/fog/milestones/v2.0.0

**Comment by plribeiro3000** [comment]
Yeah, i believe we need to merge some PR's first.

**Comment by plribeiro3000** [comment]
One of them are merged already. =)

**Comment by geemus** [comment]
Thanks!

**Comment by agis-** [comment]
I've just noticed that fog 1.27 for example depends on nokogiri ~> 1.5. So that means we *can* use it with nokogiri 1.6, and thus with Rails 4.2 too. So what was the original issue about?

**Comment by agis-** [comment]
@tnimmo @geemus @tokengeek Any insights on my comment above? Thanks!

**Comment by tnimmo** [comment]
@agis we required nokogiri => 1.6 on our project, due to other gem dependencies. Fog at the time of the creating the issue had the following dependency in gemfiles/Gemfile-ruby-1.8.7:

gem 'nokogiri', '~>1.5.11'

So I proposed bumping the version of nokogiri to 1.6, but when we checked the change logs we noticed that nokogiri 1.6 had dropped support for Ruby 1.8.7 which was still supported by Fog at the time and therefore would produce a conflict in supported Ruby versions.

@plribeiro3000 noted that Fog 2.0 would drop support for Ruby 1.8.7 and that would be the right time to upgrade the nokogiri version. We then forked fog and bumped the nokogiri version and all seem to work fine with a plan to move back to the master fog when 2.0 released.

Hope that helps
